authz/src/instrumentation.rs

```diff
@@ -0,0 +1,227 @@
+use async_trait::async_trait;
+use iox_time::{SystemProvider, TimeProvider};
+use metric::{DurationHistogram, Metric, Registry};
+
+use super::{Authorizer, Error, Permission};
+
+const AUTHZ_DURATION_METRIC: &str = "authz_permissions_duration";
```

Comment:
```suggestion
const AUTHZ_DURATION_METRIC: &str = "authz_permission_check_duration"
```

Err towards meaningful metric names rather than reflections of the code - the code changes!